### DISCUSSION TITLE
Contribution guide/custom themes?

### DISCUSSION THREAD

**Discussion by spencer-michaels** [tefkah/zotero-night#150] [Ideas]
I LOVE this plugin so much, and I want to make more themes — a proper contribution guide with how to set up the plugin/edit things like colours would be incredible! I've seen it on the to-do list in the README for ages, opening up the plugin more for community themes would be awesome. Maybe even a settings page to change visual aspects...

**Comment by tefkah** [comment]
Hey!

I would indeed love to add custom theming support!

I think I will only add it to the Zotero 7 version however, the Zotero 6 version is such a buggy mess I don't really want to mess with it anymore.

I've been making preparations to allow for this, but I'm not sure about how to go about it yet.

There's a few options I've been weighing:
1. Only allow new themes through pull requests
2. Allow users to tweak the theme through some preset variables in the settings
3. Allow users to upload their own css files through the plugin settings

I think I would prefer 1, it would be wayyyyy easier to implement, but I know as a user I would love to do some custom stuff. I unfortunately do not have a lot of time to work on the plugin (just a few hours here and there) and can't guarantee that 2 or 3 would happen in a reasonable amount of time. I did have some dreams when I started Night that it could become some platform for theming Zotero.


What are your thoughts, as a user?

**Reply by spencer-michaels** [reply]
All three sound great! I don't blame you for waiting for Zotero 7 -- I feel like 6 in general is a mess. As a user who's fairly comfortable with coding, I'd definitely make use of option 1, as would others! 2 and 3 would be friendlier to users who don't want to go into the weeds, but might be redundant if there are enough themes implemented/added via push requests. I know I'd definitely contribute (if anything, to procrastinate my work). Do you know if there's an ETA for Z7 to release out of beta? Thanks for all the work you do on the plugin, it saves my eyes during late-night grinds!
